lib/methods/timeperiodtask.cpp

```diff
@@ -31,5 +43,17 @@ Array::Ptr TimePeriodTask::EvenMinutesTimePeriodUpdate(const TimePeriod::Ptr& tp
 		}
 	}
 
+	if (Checkable::ExecuteCommandProcessFinishedHandler) {
+		double now = Utility::GetTime();
+		ProcessResult pr;
+		pr.PID = -1;
+		pr.Output = "";
+		pr.ExecutionStart = now;
+		pr.ExecutionEnd = now;
+		pr.ExitStatus = 0;
+
+		Checkable::ExecuteCommandProcessFinishedHandler("", pr);
+	}
+
```

Comment:
Damn! Sorry. Overseen that these don't have the `void *Task::ScriptFunc(const Checkable::Ptr& checkable, const CheckResult::Ptr& cr,	const Dictionary::Ptr& resolvedMacros, bool useResolvedMacros)` signature. Please revert.

On the other hand you didn't handle all with a such signature, yet.